crystalfontz/dbus/map.py

```diff
@@ -0,0 +1,625 @@
+            item.n_digits,
+            item.column,
+            item.row,
+            TemperatureUnitM.pack(item.units),
+        )
+
+
+class SetupLiveTemperatureDisplayM:
+    t: ClassVar[str] = t(IndexM, TemperatureDisplayItemM, TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def unpack(
+        slot: int,
+        item: Tuple[int, int, int, int, bool],
+        timeout: float,
+        retry_times: int,
+    ) -> Tuple[int, TemperatureDisplayItem, Optional[float], Optional[int]]:
+        return (
+            slot,
+            TemperatureDisplayItemM.unpack(item),
+            TimeoutM.unpack(timeout),
+            RetryTimesM.unpack(retry_times),
+        )
+
+
+LCD_REGISTERS: Dict[bool, LcdRegister] = {
+    bool(register.value): register for register in LcdRegister
+}
+
+
+class LcdRegisterM:
+    t: ClassVar[str] = "b"
+
+    @staticmethod
+    def unpack(register: bool) -> LcdRegister:
+        return LCD_REGISTERS[register]
+
+    @staticmethod
+    def pack(register: LcdRegister) -> bool:
+        return bool(register.value)
+
+
+class SendCommandToLcdControllerM:
+    t: ClassVar[str] = t(LcdRegisterM, ByteM, TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def unpack(
+        location: bool, data: int, timeout: float, retry_times: int
+    ) -> Tuple[LcdRegister, int, Optional[float], Optional[int]]:
+        return (
+            LcdRegisterM.unpack(location),
+            data,
+            TimeoutM.unpack(timeout),
+            RetryTimesM.unpack(retry_times),
+        )
+
+
+class ConfigureKeyReportingM:
+    t: ClassVar[str] = t(array(ByteM), array(ByteM), TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def unpack(
+        when_pressed: List[int],
+        when_released: List[int],
+        timeout: float,
+        retry_times: int,
+    ) -> Tuple[Set[int], Set[int], Optional[float], Optional[int]]:
+        return (
+            set(when_pressed),
+            set(when_released),
+            TimeoutM.unpack(timeout),
+            RetryTimesM.unpack(retry_times),
+        )
+
+
+class SpecialCharacterM:
+    t: ClassVar[str] = "t"
+
+    @staticmethod
+    def unpack(character: int) -> SpecialCharacter:
+        raise NotImplementedError("load")
+
+
+class SetSpecialCharacterDataM:
+    t: ClassVar[str] = t(IndexM, SpecialCharacterM, TimeoutM, RetryTimesM)
+
+    @staticmethod
+    def unpack(
+        index: int,
+        character: int,
+        timeout: float,
+        retry_times: int,
+    ) -> Tuple[int, SpecialCharacter, Optional[float], Optional[int]]:
+        return (
+            index,
+            SpecialCharacterM.unpack(character),
+            TimeoutM.unpack(timeout),
+            RetryTimesM.unpack(retry_times),
+        )
+
+
+class SetSpecialCharacterEncodingM:
+    t: ClassVar[str] = "sy"
+
+
+class KeypadPolledM:
+    t: ClassVar[str] = array(struct("bbb"))
+
+    @staticmethod
+    def pack(polled: KeypadPolled) -> List[Tuple[bool, bool, bool]]:
+        raise NotImplementedError("pack")
```

Comment:
Don't forget to revisit these! They're easy, I just did this out of order and haven't finished it yet.